File: solutions/cpp/difference-of-squares/1/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace difference_of_squares
{
    int square_of_sum(int num)
    {
        int res;

        res = 0;
        for (int i = 0; i <= num; i++)
            res += i;
        res *= res;
        return res;
    }
    int sum_of_squares(int num)
    {
        int res;

        res = 0;
        for (int i = 0; i <= num; i++)
            res += (i * i);
        return res;
    }
    int difference(int num)
    {
        return square_of_sum(num) - sum_of_squares(num);
    }
} // namespace difference_of_squares

File: solutions/free-pascal/difference-of-squares/1/DifferenceOfSquares.pas
unit DifferenceOfSquares;

{$mode ObjFPC}{$H+}

interface

function SquareOfSum(number : longint) : longint;
function SumOfSquares(number : longint) : longint;
function DifferenceOfSquares(number : longint) : longint;

implementation

uses SysUtils;

function SquareOfSum(number : longint) : longint;
begin

  SquareOfSum := sqr((number * (number + 1)) div 2);

end;

function SumOfSquares(number : longint) : longint;
begin
  
  result:= (number * (number + 1) * ((2 * number) + 1)) div 6;

end;

function DifferenceOfSquares(number : longint) : longint;
begin

  result := SquareOfSum(number) - SumOfSquares(number);

end;

end.
